base/MainChar.ts

```diff
@@ -273,22 +287,30 @@ export class MainChar extends Player {
         const item_obj = this.items[index];
         if (item_obj.equipped && !initialize) return;
         const item = this.info.items_list[item_obj.key_name];
+
         if (item.type in item_equip_slot && this.equip_slots[item_equip_slot[item.type]] !== null) {
             this.unequip_item(this.equip_slots[item_equip_slot[item.type]].index);
         }
         if (item.type in item_equip_slot) {
             this.equip_slots[item_equip_slot[item.type]] = item_obj;
         }
+
         item_obj.equipped = true;
         for (let i = 0; i < item.effects.length; ++i) {
             this.add_effect(item.effects[i], item);
         }
+
         this.update_attributes();
         this.update_elemental_attributes();
         if (item.type === item_types.ABILITY_GRANTOR) {
             this.equipped_abilities.push(item.granted_ability);
             this.update_abilities();
         }
+
+        if (item.type === item_types.CLASS_CHANGER) {
```

Comment:
please use `else if`, so we don't need to check the `item.type` again. Also, after updating the class, we need to update the attributes, so I suggest moving `this.update_attributes();` from line 303 to just after these if elses checks.